seastats/stats.py

```diff
@@ -214,28 +219,53 @@ def get_stats(
-        "rms_95": get_rms_quantile(sim, obs, 0.95, round),
-        "sim_mean": np.round(sim.mean(), round),
-        "obs_mean": np.round(obs.mean(), round),
-        "sim_std": np.round(sim.std(), round),
-        "obs_std": np.round(obs.std(), round),
-        "nse": get_nse(sim, obs, round),
-        "lamba": get_lambda(sim, obs, round),
-        "cr": get_corr(sim, obs, round),
-        "cr_95": get_corr_quantile(sim, obs, 0.95, round),
-        "slope": slope,
-        "intercept": intercept,
-        "slope_pp": slopepp,
-        "intercept_pp": interceptpp,
-        "mad": get_mad(sim, obs, round),
-        "madp": get_madp(sim, obs, round),
-        "madc": get_madc(sim, obs, round),
-        "kge": get_kge(sim, obs, round),
-    }
+
+    def should_calculate(metric_name):
+        return metrics is None or metric_name in metrics_list
+
+    metrics_list = [metrics] if isinstance(metrics, str) else metrics
+    version_stat = {}
+
+    # Calculate each metric if it should be calculated
+    if should_calculate("bias"):
+        version_stat["bias"] = get_bias(sim, obs, round)
+    if should_calculate("rmse"):
+        version_stat["rmse"] = get_rmse(sim, obs, round)
+    if should_calculate("rms"):
+        version_stat["rms"] = get_rms(sim, obs, round)
+    if should_calculate("rms_95"):
+        version_stat["rms_95"] = get_rms_quantile(sim, obs, 0.95, round)
+    if should_calculate("sim_mean"):
+        version_stat["sim_mean"] = np.round(sim.mean(), round)
+    if should_calculate("obs_mean"):
+        version_stat["obs_mean"] = np.round(obs.mean(), round)
+    if should_calculate("sim_std"):
+        version_stat["sim_std"] = np.round(sim.std(), round)
+    if should_calculate("obs_std"):
+        version_stat["obs_std"] = np.round(obs.std(), round)
+    if should_calculate("nse"):
+        version_stat["nse"] = get_nse(sim, obs, round)
+    if should_calculate("lamba"):
+        version_stat["lamba"] = get_lambda(sim, obs, round)
+    if should_calculate("cr"):
+        version_stat["cr"] = get_corr(sim, obs, round)
+    if should_calculate("cr_95"):
+        version_stat["cr_95"] = get_corr_quantile(sim, obs, 0.95, round)
+    if should_calculate("slope") or should_calculate("intercept"):
+        slope, intercept = get_slope_intercept(sim, obs)
+        version_stat["slope"] = slope
+        version_stat["intercept"] = intercept
+    if should_calculate("slope_pp") or should_calculate("intercept_pp"):
+        slopepp, interceptpp = get_slope_intercept_pp(sim, obs)
+        version_stat["slope_pp"] = slopepp
+        version_stat["intercept_pp"] = interceptpp
+    if should_calculate("mad"):
+        version_stat["mad"] = get_mad(sim, obs, round)
+    if should_calculate("madp"):
+        version_stat["madp"] = get_madp(sim, obs, round)
+    if should_calculate("madc"):
+        version_stat["madc"] = get_madc(sim, obs, round)
+    if should_calculate("kge"):
+        version_stat["kge"] = get_kge(sim, obs, round)
+
```

Comment:
For each metric you are making two function calls instead of one (`should_calculate() + get_*()` VS just `get_*()`). 

In this specific case, performance should not be affected because the `get_*()` are probably dominating the runtime. That being said it is possible to refactor and and avoid the `should_calculate()`. 

A common way to avoid the consecutive `ifs` is to define a dictionary that maps the metrics name to the function calls. For example:

```python
_METRIC_TO_FUNC = {
    "mad": get_mad,
    "madp": get_madp,
    "madc": get_madc,
    "kge": get_kge,
}

version_stat = {}
for metric_name in metric_list:
    func = _METRIC_TO_FUNC[metric_name]
    version_stat[metric_name] = func(sim, obs, round)
```
The caveat here is that all the functions need to have the same signature. You can circumvent this by adding `*args, **kwargs` in all the functions, but it is a bit ugly IMHV.

Another way is to use the relatively new `match` statement which has beed added exactly for this use case. It should be something like this:
``` python
for metric_name in metrics_list:
    match metric_name:
        case "mad":
            version_stat[metric_name] = get_mad(sim, obs, round)
        case "madp":
            version_stat[metric_name] = get_madp(sim, obs, round)
        case "madc":
            version_stat[metric_name] = get_madc(sim, obs, round)
        case "kge":
            version_stat[metric_name] = get_kge(sim, obs, round)
````
In this specific case I would go for the match case, but it is only supported after 3.10

Reply:
let's go for the `match` statement then. python 3.8 is end-of-life already and 3.9 ends in Oct 2025.

Reply:
I updated CI for python versions above 3.10 included
It's indeed more elegant this way, thanks.